src/controllers/wallet/atomic-swap/tx-proposal.controller.js

```diff
@@ -30,86 +32,114 @@ async function buildTxProposal(req, res) {
 
   const network = req.wallet.getNetworkObject();
 
-  const inputs = req.body.inputs || [];
-  const outputs = req.body.outputs || [];
-  const sendTokens = req.body.send_tokens || [];
-  const receiveTokens = req.body.receive_tokens || [];
+  const sendTokens = req.body.send || { tokens: [] };
+  const receiveTokens = req.body.receive || { tokens: [] };
   const changeAddress = req.body.change_address || null;
   const partialTx = req.body.partial_tx || null;
   let markAsSelected = true;
   if (req.body.lock !== undefined) {
     markAsSelected = req.body.lock;
   }
+  const utxos = [];
 
-  if (inputs.length === 0
-    && outputs.length === 0
-    && sendTokens.length === 0
-    && receiveTokens.length === 0
-  ) {
+  if (sendTokens.tokens.length === 0 && receiveTokens.tokens.length === 0) {
     res.status(400).json({ success: false, error: 'Should have at least one operation' });
     return;
   }
 
   const proposal = partialTx
     ? PartialTxProposal.fromPartialTx(partialTx, network) : new PartialTxProposal(network);
 
-  for (const input of inputs) {
+  if (sendTokens.utxos && sendTokens.utxos.length > 0) {
     try {
-      const txData = await new Promise((resolve, reject) => {
-        txApi.getTransaction(input.txId, data => resolve(data))
-          .catch(err => reject(err));
-      });
+      const currentTs = dateFormatter.dateToTimestamp(new Date());
+      for (const utxo of sendTokens.utxos) {
+        const txData = await new Promise((resolve, reject) => {
+          txApi.getTransaction(utxo.txId, data => resolve(data))
+            .catch(err => reject(err));
+        });
 
-      if (!txData.success) {
-        throw new Error(`Utxo for transaction ${input.txId} and input ${input.index} not found`);
-      }
+        if (!txData.success) {
+          throw new Error(`Utxo for transaction ${utxo.txId} and index ${utxo.index} not found`);
+        }
 
-      const txout = txData.tx.outputs[input.index];
-      proposal.addInput(
-        req.wallet,
-        txout.txId,
-        txout.index,
-        txout.value,
-        txout.decoded.address,
-        {
-          markAsSelected,
-          token_data: txout.token_data,
-          token: txout.token,
-        },
-      );
+        let heightLocked = false;
+        if (txData.tx.height) {
+          const blocksMined = (oldWallet.getNetworkHeight() - txData.tx.height);
+          heightLocked = blocksMined < oldWallet.getRewardLockConstant();
+          if (heightLocked) {
+            throw new Error(`Utxo for transaction ${utxo.txId} and index ${utxo.index} \
+is a block reward and require ${oldWallet.getRewardLockConstant() - blocksMined} more block(s) \
+to be mined before it can be spent.`);
+          }
+        }
+
+        const txout = txData.tx.outputs[utxo.index];
+
+        const addressIndex = req.wallet.getAddressIndex(txout.decoded.address);
+        const addressPath = addressIndex ? `m/44'/${HATHOR_BIP44_CODE}'/0'/0/${addressIndex}` : '';
```

Comment:
What about using the account path available on the lib's `constants`? 
```suggestion
        const addressPath = addressIndex ? `${P2PKH_ACCT_PATH}'/0/${addressIndex}` : '';
```

Reply:
Changed to use the address path helper from the lib.